src/chains/opStack/actions/proveWithdrawal.ts

```diff
@@ -0,0 +1,147 @@
+  UnionOmit<
+    FormattedTransactionRequest<_derivedChain>,
+    | 'accessList'
+    | 'data'
+    | 'from'
+    | 'gas'
+    | 'gasPrice'
+    | 'to'
+    | 'type'
+    | 'value'
+  >
+> &
+  GetAccountParameter<account, Account | Address> &
+  GetChainParameter<chain, chainOverride> &
+  GetContractAddressParameter<_derivedChain, 'portal'> & {
+    /**
+     * Gas limit for transaction execution on the L1.
+     * `null` to skip gas estimation & defer calculation to signer.
+     */
+    gas?: bigint | null
+    l2OutputIndex: bigint
+    outputRootProof: {
+      version: Hex
+      stateRoot: Hex
+      messagePasserStorageRoot: Hex
+      latestBlockhash: Hex
+    }
+    withdrawalProof: readonly Hex[]
+    withdrawal: {
+      data: Hex
+      gasLimit: bigint
+      nonce: bigint
+      sender: Address
+      target: Address
+      value: bigint
+    }
+  }
+export type ProveWithdrawalReturnType = Hash
+export type ProveWithdrawalErrorType =
+  | EstimateContractGasErrorType
+  | WriteContractErrorType
+  | ErrorType
+
+/**
+ * Proves a withdrawal that occurred on an L2. Used in the Withdrawal flow.
+ *
+ * - Docs: https://viem.sh/op-stack/actions/proveWithdrawal.html
+ *
+ * @param client - Client to use
+ * @param parameters - {@link ProveWithdrawalParameters}
+ * @returns The prove transaction hash. {@link ProveWithdrawalReturnType}
+ *
+ * @example
+ * import { createWalletClient, http } from 'viem'
+ * import { mainnet, optimism } from 'viem/chains'
+ * import { proveWithdrawal } from 'viem/op-stack'
+ *
+ * const walletClientL1 = createWalletClient({
+ *   chain: mainnet,
+ *   transport: http(),
+ * })
+ *
+ * const request = await proveWithdrawal(walletClientL1, {
+ *   l2OutputIndex: 4529n,
+ *   outputRootProof: { ... },
+ *   targetChain: optimism,
+ *   withdrawalProof: [ ... ],
+ *   withdrawal: { ... },
+ * })
+ */
+export async function proveWithdrawal<
+  chain extends Chain | undefined,
+  account extends Account | undefined,
+  chainOverride extends Chain | undefined = undefined,
+>(
+  client: Client<Transport, chain, account>,
+  parameters: ProveWithdrawalParameters<chain, account, chainOverride>,
+): Promise<ProveWithdrawalReturnType> {
+  const {
+    account,
+    chain = client.chain,
+    gas,
+    l2OutputIndex,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+    nonce,
+    outputRootProof,
+    targetChain,
+    withdrawalProof,
+    withdrawal,
+  } = parameters
+
+  const portalAddress = (() => {
+    if (parameters.portalAddress) return parameters.portalAddress
+    if (chain) return targetChain!.contracts.portal[chain.id].address
+    return Object.values(targetChain!.contracts.portal)[0].address
+  })()
+
+  const args = {
+    account,
+    abi: portalAbi,
+    address: portalAddress,
+    chain,
+    functionName: 'proveWithdrawalTransaction',
+    args: [withdrawal, l2OutputIndex, outputRootProof, withdrawalProof],
+    gas,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+    nonce,
+  } as any
```

Comment:
```suggestion
  } satisfies ArgsType as any
```
Can still get some typechecking here if you do this I think